docs/static/fb-ls-kafka-example.asciidoc

```diff
@@ -0,0 +1,140 @@
+step because {filebeat} needs to create the index template in {es} and
+load the sample dashboards into {kib}. 
++
+After the template and dashboards are loaded, you'll see the message `INFO
+{kib} dashboards successfully loaded. Loaded dashboards`.
+
+. Run the `modules enable` command to enable the modules that you want to run.
+For example:
++
+[source,shell]
+----------------------------------------------------------------------
+filebeat modules enable system
+----------------------------------------------------------------------
++
+You can further configure the module by editing the config file under the
+{filebeat} `modules.d` directory. For example, if the log files are not in the
+location expected by the module, you can set the `var.paths` option.
+
+. Run the `setup` command with the `--pipelines` and `--modules` options
+specified to load ingest pipelines for the modules you've enabled. This step
+also requires a connection to {es}. If you want use a {ls} pipeline instead of
+ingest node to parse the data, skip this step.
++
+[source,shell]
+----------------------------------------------------------------------
+filebeat setup --pipelines --modules system
+----------------------------------------------------------------------
+
+. Configure {filebeat} to send log lines to Kafka. To do this, in the
++filebeat.yml+ config file, disable the {es} output by commenting it out, and
+enable the Kafka output. For example:
++
+[source,yaml]
+-----
+#output.elasticsearch:
+  #hosts: ["localhost:9200"]
+output.kafka:
+  hosts: ["kafka:9092"]
+  topic: "filebeat"
+  codec.json:
+    pretty: false
+-----
+
+. Start {filebeat}. For example:
++
+[source,shell]
+----------------------------------------------------------------------
+filebeat -e
+----------------------------------------------------------------------
++
+{filebeat} will attempt to send messages to {ls} and continue until {ls} is
+available to receive them.
++
+NOTE: Depending on how you've installed {filebeat}, you might see errors
+related to file ownership or permissions when you try to run {filebeat} modules.
+See {beats-ref}/config-file-permissions.html[Config File Ownership and Permissions]
+in the _Beats Platform Reference_ if you encounter errors related to file
+ownership or permissions.
++
+See {filebeat-ref}/filebeat-starting.html[Starting {filebeat}] for more info.
+
+==== Create and start the {ls} pipeline
+
+. On the system where {ls} is installed, create a {ls} pipeline configuration
+that reads from a Kafka input and sends events to an {es} output:
++
+--
+[source,shell]
+-----
+input {
+  kafka {
+    bootstrap_servers => "myhost:9092"
+    topics => ["filebeat"]
+    codec => json
+  }
+}
+
+output {
+  elasticsearch {
+    hosts: ["https://myEShost:9200"]
+    manage_template => false
+    index => "%{[@metadata][beat]}-%{[@metadata][version]}-%{+YYYY.MM.dd}"
+    pipeline => "%{[@metadata][pipeline]}" <1>
```

Comment:
In my experience you need to put a conditional around the `elasticsearch` output to check if `@metadata.pipeline` is set. Otherwise if you have any data that doesn't include a @metadata.pipeline you will get errors sending to Elasticsearch because logstash sets an empty `pipeline: ""` value in the _bulk metadata.

```
if [@metadata][pipeline] {
  // es output with pipeline
} else {
  // es output w/o pipeline
}
```